tfhe/src/core_crypto/gpu/vec.rs

```diff
@@ -41,7 +41,9 @@ impl<T: Numeric> CudaVec<T> {
     ///   not be dropped until stream is synchronised
     pub unsafe fn from_async(h_data: &Vec<T>, stream: &CudaStream) -> Self {
         let mut d_data = stream.malloc_async(h_data.len() as u32);
-        stream.copy_to_gpu_async(&mut d_data, h_data.as_slice());
+        if !h_data.is_empty() {
```

Comment:
I had to add this condition otherwise we trying to copy data from an invalid pointer.

Reply:
ah yes, this is a rust thing when representing slices, if it's empty the pointer can be any well aligned pointer for the type even if there is no memory allocation there